xls/contrib/mlir/IR/xls_ops.cc

```diff
@@ -390,6 +390,32 @@ Type getI1TypeOf(Type type) {
 
 namespace mlir::xls {
 
+LogicalResult LiteralOp::verifyRegions() {
+  Block& b = getInitializerBlock();
+  // Verify return op & typing:
+  auto ret = dyn_cast<YieldOp>(b.getTerminator());
+  if (!ret) {
+    return emitOpError(
+        "initializer region must be terminated by an xls.yield op");
+  }
+  if (ret->operand_type_begin() == ret.operand_type_end()) {
+    return emitOpError("initializer region cannot return void");
+  }
+  if (*(ret->operand_type_begin()) != getType()) {
+    return emitOpError("initializer region type ")
+           << *ret.operand_type_begin() << " does not match literal type "
+           << getType();
+  }
+  // Verify that operations inside are allowed inside a literal region:
+  for (Operation& op : b) {
```

Comment:
If you make it a trait, you should be able to use llvm all\_of kind of check here with trait checking.